### ISSUE TITLE
Improve DNS checks to allow DNS servers not working temporary

### ISSUE THREAD

**Issue by MrGeneration** [modoboa/modoboa#2038]
# Impacted versions

* Modoboa: 1.16.1
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce

Force your DNS resolver to be unavailable for a moment (for whatever reason).

# Current behavior

Modoboa regulary tries to resolve all relevant DNS entries for configured domains if enabled. 
It will, in case e.g. a MX record no longer can be found, send a mail to the domain owner to inform about this issue.

![image](https://user-images.githubusercontent.com/6549061/96621148-8eb7ac00-1308-11eb-94a9-fc3e7c9e1a02.png)

# Expected behavior

If Modoboa is unable to query the DNS server for a moment it should temporarily ignore this issue (maybe change the record from green to yellow to indicate a temporary issue) and maybe inform the instance owner. The logfile states a resolution timeout:

![image](https://user-images.githubusercontent.com/6549061/96621394-db02ec00-1308-11eb-8e80-6d861e003d5e.png)

Issue #1023 technically would be very useful in such an situation as well.

# TL;DR

If DNS resolution fails inform instance owner and do not thread this situation as non existent DNS record.

**Comment by tonioo** [comment]
@MrGeneration Actually you can already disable tests for a specific domain. Is it not enough?

**Comment by MrGeneration** [comment]
While this is true, this was temporary affecting 50% of my domains due a temporary route issue to the affected DNS servers.
So sadly, no this wouldn't be an option.

Personally I think that the DNS check should be robust enough to know the difference between a simple timeout or `NXDOMAIN`. Because this would also allow you to throw different errors.

I really love the domain check functionality and am using it where ever possible.

**Comment by MrGeneration** [comment]
I still believe that modoboa should handle not reachable DNS servers better. :x
